docs/sapling-comparison.md

```diff
@@ -0,0 +1,70 @@
+  user explicitly tells the tool when to create a commit and which files to
+  include. When using jj, the working copy
+  is [automatically snapshotted by every command](working-copy.md). New files
+  are automatically tracked and deleted files are automatically untracked. This
+  has several advantages:
+    * The working copy is effectively backed up every time you run a command.
+    * No commands fail because you have changes in the working copy ("abort: 1
+      conflicting file changes: ..."). No need for `sl shelve`.
+    * Simpler and more consistent CLI because the working copy is treated like
+      any other commit.
+* **Conflicts:** Like most VCSs, Sapling requires the user to
+  resolve conflicts before committing. jj lets
+  you [commit conflicts](conflicts.md). Note that it's a representation of the
+  conflict that's committed, not conflict markers (`<<<<<<<` etc.). This also
+  has several advantages:
+    * Merge conflicts won't prevent you from checking out another commit.
+    * You can resolve the conflicts when you feel like it.
+    * Rebasing descendants always succeeds. Like jj, Sapling automatically
+      rebases, but it will fail if there are conflicts.
+    * Merge commits can be rebased correctly (Sapling sometimes fails).
+    * You can rebase conflicts and conflict resolutions.
+* **Undo:** jj's undo is powered by [the operation log](operation-log.md), which
+  records how the repo has changed over time. Sapling has a similar feature
+  with its [MetaLog](https://sapling-scm.com/docs/internals/metalog).
+  They seem to provide similar functionality, but jj also exposes the log to the
+  user via `jj op log`, so you can tell how far back you want to go back.
+  Sapling has `sl debugmetalog`, but that seems to show the history of a single
+  commit, not the whole repo's history. Thanks to jj snapshotting the working
+  copy, it's possible to undo changes to the working copy. For example, if
+  you `jj undo` a ` jj commit`, `jj diff` will show the same changes as
+  before `jj commit`, but if you `sl undo` a `sl commit`, the working copy will
+  be clean.
+* **Git interop:** Sapling supports cloning, pushing, and pulling from a remote
+  Git repo. jj also does, and it also supports sharing a working copy with a Git
+  repo, so you can use `jj` and `git` interchangeably in the same repo.
+* **Polish:** Sapling is much more polished and feature-complete. For example,
+  jj has no `blame/annotate` or `bisect` commands, and also no copy/rename
+  support. Sapling also has very nice web UI
+  called [Interactive Smartlog](https://sapling-scm.com/docs/addons/isl), which
+  lets you drag and drop commits to rebase them, among other things.
+* **Forge integrations:** Sapling has `sl pr submit --stack`, which lets you
+  push a stack of commits as separate GitHub PRs (or GitLab MRs, since it simply
+  creates branches?). You will then have to manually add the base branch for
```

Comment:
`sl pr` uses GitHub's API thus does not work with GitLab. (`sl pr submit` does not just create branches.)

Reply:
Makes sense. Is there a way to just create branches on an non-GitHub remote? For now, I've updated the text to say that only GitHub is supported, but I'm happy to correct it if Sapling supports pushing branches. Maybe `sl push` can do that?

Reply:
> Is there a way to just create branches on an non-GitHub remote?

Yes, I do this all the time (with GitHub). `sl push -r . --to master` pushes the current commit (`.`) to a remote branch called `master`, for example.

Sapling can also manage remote branches. For example, `sl push --delete tempbranch` behaves like `git push origin :refs/heads/tempbranch`.

> For now, I've updated the text to say that only GitHub is supported

I thought this section is about *forge-specific integrations*, not general Git server support. Sapling does support pushing to and pulling from Git servers.

Reply:
> I thought this section is about _forge-specific integrations_, not general Git server support. Sapling does support pushing to and pulling from Git servers.

Ah, good point, "integrations" wasn't correct for what I meant. I've changed it to "Forge workflows".